### ISSUE TITLE
misdetedcting explicit combination of licenses

### ISSUE THREAD

**Issue by jonassmedegaard** [nexB/scancode-toolkit#929]
ScanCode seems (from simply looking at YAML data for tests) to not recognize licenses explicitly tied together.

These lack recognition of explicit "MPL-1.1 or GPL-2+ or LGPL-2.1+":
  * tests/ScanCode/tests/licensedcode/data/licenses/apache-2.0_and_bsd-new_and_gpl-2.0-plus_and_lgpl-2.1-plus_and_mpl-1.1_and_other.yml
  * tests/ScanCode/tests/licensedcode/data/licenses/gpl-2.0-plus_and_gpl-2.0-plus_and_lgpl-2.1-plus_and_mpl-1.1_and_other
  * tests/ScanCode/tests/licensedcode/data/licenses/gpl-2.0-plus_and_lgpl-2.1-plus_and_mpl-1.1

This one lacks recognition of explicit "Apache-2.0 or GPL-2":
  * tests/ScanCode/tests/licensedcode/data/licenses/

There are more, but above are the only ones that Licensecheck succeed currently.

**Comment by pombredanne** [comment]
@jonassmedegaard Thank you ++ for these great finds!

**Comment by pombredanne** [comment]
FWIW, we should detect all these alright but these need to be surfaced in the output with  #74

**Comment by pombredanne** [comment]
Expressions are now fully supported and available in most all recent tags including 2.9.7
Thanks!
For reference in https://github.com/nexB/scancode-toolkit/blob/2148aedc3158ab9d0eec98ff2fe72e489c3de5e0/tests/licensedcode/data/licenses/apache-2.0_and_bsd-new_and_gpl-2.0-plus_and_lgpl-2.1-plus_and_mpl-1.1_and_other.yml#L2
we detect these:
```
license_expressions:
    - mpl-1.1 OR gpl-2.0-plus OR lgpl-2.1-plus
    - mpl-1.1 OR gpl-2.0-plus OR lgpl-2.1-plus
    - lgpl-2.1-plus
    - bsd-new
    - bsd-new
    - apache-2.0
    - ms-pl
expected_failure: yes
```
where the two first are expressions alright
